pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -436,6 +447,15 @@ func (h *Health) checkObservabilityComponents(
 		}
 	}
 
+	for _, extensionCondition := range extensionConditions {
+		if extensionCondition.ExtensionType == "shoot-networking-problemdetector" {
```

Comment:
We don't hard-code extension names in this repository. Instead, you should make use of https://github.com/gardener/gardener/blob/master/docs/extensions/healthcheck-library.md. Concretely, https://github.com/gardener/gardener-extension-shoot-networking-problemdetector/blob/0cd9bd0dedeb59fd01fc12aaeb71a8d8aff6895c/pkg/controller/healthcheck/add.go#L43 should be adapted to report a health check for the `ShootObservabilityComponentsHealthy` condition type (similar to https://github.com/gardener/gardener-extension-provider-aws/blob/63677ab3c32c454c77ccc90c77e688afbb24cba5/pkg/controller/healthcheck/add.go#L67-L70 for example) by using https://github.com/gardener/gardener/blob/086280a89b5ed79e28a948e626c8d5815e221731/extensions/pkg/controller/healthcheck/general/managed_resource.go#L35.

Reply:
Yeah, created PR in network-problem-detector extension. https://github.com/gardener/gardener-extension-shoot-networking-problemdetector/pull/119